tensordict/nn/common.py

```diff
@@ -276,18 +276,106 @@ def wrapper(_self, *args: Any, **kwargs: Any) -> Any:
         return wrapper
 
 
+class _OutKeysSelect:
+    def __init__(self, out_keys):
+        self.out_keys = out_keys
+        self._initialized = False
+
+    def _init(self, module):
+        if self._initialized:
+            return
+        self._initialized = True
+        self.module = module
+        module.out_keys = self.out_keys
+
+    def __call__(
+        self,
+        module: TensorDictModuleBase,
+        tensordict_in: TensorDictBase,
+        tensordict_out: TensorDictBase,
```

Comment:
I think the typing here is confusing as we sometimes expect these to be tuples of tensors right?